src/cli/api/local_api.py

```diff
@@ -123,19 +127,122 @@ def list_applications(self) -> List[ApplicationType]:
+                if not schema_valid:
+                    error_list.append(ve)
+            else:
+                error_list.append(message)
+
+        if os.path.isfile(network_path):
+            json_valid, message = validate_json_string(network_path)
+            if json_valid:
+                schema_valid, ve = validate_json_schema(network_path, schema_app_path / "appconfig_network.json")
+                if not schema_valid:
+                    error_list.append(ve)
+            else:
+                error_list.append(message)
+
+        if os.path.isfile(result_path):
+            json_valid, message = validate_json_string(result_path)
+            if json_valid:
+                schema_valid, ve = validate_json_schema(result_path, schema_app_path / "appconfig_result.json")
+                if not schema_valid:
+                    error_list.append(ve)
+            else:
+                error_list.append(message)
+
+        return error_list
+
+    def __is_structure_valid(self, application_name: str) -> List[str]:
+        app_dir_path = Path(self.__config_manager.get_application(application_name)['path'])
+        app_config = app_dir_path / "config"
+        app_src = app_dir_path / "src"
+        error_list = []
+
+        # Validate that the config, src and MANIFEST.ini structure is correct
+        if not os.path.exists(app_config) or \
+           not os.path.exists(app_src) or \
+           not os.path.isfile(app_dir_path / "MANIFEST.ini"):
+            error_list.append(f"In file {app_dir_path}: should contain a 'MANIFEST.ini', 'src' directory and 'config' "
+                              f"directory")
+
+        # Check if the config directory is complete
+        if os.path.exists(app_config):
+            if not os.path.isfile(app_config / "application.json") or \
+               not os.path.isfile(app_config / "network.json") or \
+               not os.path.isfile(app_config / "result.json"):
+                error_list.append(f"In file {app_config}: should contain the files 'network.json', 'result.json' and "
+                                  f"'application.json'")
+
+        # Check if the python files match the roles in application.json and vice versa
+        if os.path.exists(app_src) and os.path.isfile(app_config / "application.json"):
+            valid, message = validate_json_string(app_config / "application.json")
+            if valid:
+                application = read_json_file(app_config / "application.json")
+                config_roles = []
+
+                for item in application:
+                    for role in item["roles"]:
+                        config_roles.append(role)
+
+                config_roles = ['app_' + item + '.py' for item in config_roles]
+                application_roles = os.listdir(app_src)
+
+                if not all(item in config_roles for item in application_roles):
```

Comment:
I can image there are temporary files in app_src that are not app_<role>.py files, for example readme.txt. We get an error then?

Reply:
I removed this check because this way other files are allowed in the 'src' directory other than the role files